aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -98,6 +98,17 @@ static void neg_kernel(TensorIterator& iter) {
   });
 }
 
+static void sign_kernel(TensorIterator& iter){
+  AT_DISPATCH_ALL_TYPES(iter.dtype(), "sign_cpu", [&]() {
+    cpu_kernel(
```

Comment:
Consider adding cpu_kernel_vec

Reply:
Do I have to provide Vec256 support for that ? (Not quite sure to see the actual meaning of that structure)

Reply:
No -- you should be able to build it from the same primitive operations `<` and `-`.

Reply:
Perfect, will do asap.

Reply:
Hum , doesn't seem that trivial, I'll have to investigate a little bit more. With the almost same approach (scalar_t -> Vec256<scalar_t>) I get the following output: 

```python
>>> torch.randn(20).sign()
tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, -1., -1.,  1., -1.])
```

Reply:
Ok, for any tensor shape that is not power of 2, then it seems to fill with some random value at the end ... 

Do you have any tips why this is happening ?

Reply:
If it's not straightforward, I think we can defer the cpu_kernel_vec bit. But some basic before/after timing numbers would be good. You can use "%timeit" in iPython: 

```python
import torch
x = torch.randn(1000000)  # choose some reasonable sizes
%timeit torch.sign(x)
```


```python
import torch
x = torch.randn(1000000, device='cuda')  # choose some reasonable sizes
%timeit torch.sign(x); torch.cuda.synchronize()  # cuda needs synchronize for proper timing
```

Reply:
I dis push the vectorized implementation this morning, just had to dig a little but more than copy/past the cpu backend.

Im waiting for the master compilation to finish on my system to include "before"